src/v2/client/introspect.js

```diff
@@ -12,10 +12,23 @@
 
 import idx from '@okta/okta-idx-js';
 
-export async function introspect(settings) {
+export async function introspect(settings, appState) {
   const domain = settings.get('baseUrl');
   const stateHandle = settings.get('stateToken');
   const version = settings.get('apiVersion');
 
+  // For certain flows, we need to generate a device fingerprint
+  // to determine if we need to send a "New Device Sign-on Notification".
+  // In the future, this should be handled by okta-auth-js
+  idx.client.interceptors.request.use(requestConfig => {
+    const fingerprintRequestUrls = [
+      `${domain}/idp/idx/identify`,
+    ];
+    const fingerprint = appState.get('deviceFingerprint');
+    if (fingerprint && fingerprintRequestUrls.includes(requestConfig.url)) {
```

Comment:
what about even remove the url check?

Reply:
We actually need to keep this here - as we only want to **add the header** for this specific endpoint. Our presentation layer only notifies our app state that we have a device fingerprint, not to act on it.

Reply:
I was thinking there is no way (assume bug free) for `fingerprint` is defined but not at `identify` step?
Let's see tomorrow we want finger print for `/enroll`, we 'll need to modify Enroll View as well as add URL check. When 2 changes happen at same time, usually it's code smell ?

Reply:
This is a valid concern, and one we brought up with our server-side team. Luckily, `/identify` will be the **only** path that we're looking for this device fingerprint on OIE, as the backend will [include the fingerprint in the auth state](https://oktawiki.atlassian.net/wiki/spaces/~yuming.cao/pages/1845726231/New+device+sign-in+notification+checking+logic+on+OIE?focusedCommentId=1917097435#comment-1917097435) for all subsequent requests.

We _could_ find a way to clean up this code even more by moving the fingerprinting generation outside of the Identify view into this handler, but this seems like more of a view-level responsibility as we don't have access to the `$el` here.

What do you think?